suites/tipset/rewards_penalties.go

```diff
@@ -208,6 +209,39 @@ func TestMinerRewardsAndPenalties(t *testing.T, factory state.Factories) {
 		td.AssertBalance(builtin.BurntFundsActorAddr, big.Add(halfBalance, gasPenalty))
 	})
 
+	t.Run("insufficient gas to cover return value", func(t *testing.T) {
+		td := builder.Build(t)
+		defer td.Complete()
+
+		miner := td.ExeCtx.Miner
+		tb := drivers.NewTipSetMessageBuilder(td)
+		bb := drivers.NewBlockBuilder(td, td.ExeCtx.Miner)
+
+		alice, _ := td.NewAccountActor(drivers.BLS, acctDefaultBalance)
+
+		gasPrice := int64(1)
+		gasLimit := int64(458)
+		// message and operation cost is 458 gas units
+		// return value costs 56 units of gas, the above isn't enough to cover return value.
+		retval := chain.MustSerialize(&miner_spec.GetControlAddressesReturn{
+			Owner:  td.StateDriver.BuiltinMinerInfo().OwnerID,
+			Worker: td.StateDriver.BuiltinMinerInfo().WorkerID,
+		})
+		bb.WithBLSMessageAndCodeAndReturn(
+			td.MessageProducer.MinerControlAddresses(miner, alice,
+				nil, chain.Nonce(0), chain.GasPrice(gasPrice), chain.GasLimit(gasLimit)),
+			exitcode.SysErrOutOfGas,
```

Comment:
Flagging for discussion: We cannot assert the error code `SysErrOutOfGas` was returned due to lack of gas for covering the return value. This makes the test somewhat unreliable. Thoughts?

Reply:
Yes. We don't need to hardcode a value here. Send the message once with a high gas limit and record what the actual gas cost was. Then send it again with a gas limit one lower than that.

It wouldn't test the right thing if the return value was free, but I think good enough.